### PR TITLE
Rail dev

### PR DESCRIPTION
**Pull request by eacharles**
These changes allow ceci to use stages from namespace modules.   

1. Adding `module_name` as a member to the StageExecutionConfig object, storing it in the configuration files, etc.   These changes are backwards compatible.  If `module_name` is not present in the file it will simple default to None.
2. Having `PipelineStage.get_stage` optionally take `module_name` as an argument.  If it is present, then if the Stage wasn't loaded when the top-level module was loaded (i.e., b/c it is a namespace), then it will load the specific module when `get_stage` is called.
3. Changing the logic in `PipelineStage.generate_command` to treat regular packages (e.g., `python -m foo ClassName`) and namespace packages (e.g., `python -m ceci foo.bar.ClassName`) slightly differently.  
4. Changing the logic in `PipelineStage.main` to treat full class names (e.g., foo.bar.ClassName) differently from short class names (e.g., ClassName).   In the former case 'foo.bar' will be imported, in the later case, the behavior is unchanged.  
5. Added __main__.py to ceci as a way to provide a __main__ for namespace packages.

To test this I added 1 test in `tests/test_main.py` which runs the full example pipeline, but from a namespace packages.   This required added `ceci_example_namespace` with example code, and `tests/test_namespace.yml` and `tests/config_nm.yml` with the configuration to run it.

Finally, I updated .pylintrc to disable a number of warnings related to old flags.

### CODE REVIEW

File: ceci/stage.py [hunk 1]
```diff
@@ -1284,7 +1298,15 @@ def generate_command(
         module = cls.get_module()
         module = module.split(".")[0]
 
-        flags = [cls.name]
+        if sys.modules[module].__file__:
```
Comment (joezuntz) [inline ceci/stage.py:1301]: I don't quite understand this bit - can you explain in a comment?  When is the `__file__` attribute of a module non-empty?

**Review by joezuntz — CHANGES_REQUESTED** [review]

**Comment by joezuntz** [comment]
Looks good! Just one quick request for clarification.

**Comment by eacharles** [comment]
If it is a package, __file__ is set, if it is a namespace it is not set, but rather uses __path__ to resolve searches.

Does that clarify?

-e

**Comment by eacharles** [comment]
I can add a comment to that effect in the code once I get to work.
-e

File: ceci/stage.py (hunk 1 above)
Comment (eacharles) [reply ceci/stage.py:1301]: Ok, I clarified the comment

**Review by joezuntz — APPROVED** [review]

**Pull request merged**